### PR TITLE
Expose temporary channel ID and user channel ID pre-funding

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Its pretty hard for users to track a channel pre-funding to ChannelClosed especially when our counterparty closed the channel due to a parameter mismatch. This resolves that with more data in the relevant events.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
> If we're gonna keep a user id around, it might be helpful to have the abilitty to set a channel's user id (just once), so once an iincoming channel is persisted to some local database, it's easier to keep track of.

Yea, there's a few things we should probably consider for the user id, since its not new or (materially) changed in this PR, lets leave it for a followup and figure out what we want to do with it in #1125.

File: lightning/src/ln/channelmanager.rs [hunk 1]
```diff
@@ -1267,22 +1267,30 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 
 	/// Creates a new outbound channel to the given remote node and with the given value.
 	///
-	/// user_id will be provided back as user_channel_id in FundingGenerationReady events to allow
-	/// tracking of which events correspond with which create_channel call. Note that the
-	/// user_channel_id defaults to 0 for inbound channels, so you may wish to avoid using 0 for
-	/// user_id here. user_id has no meaning inside of LDK, it is simply copied to events and
-	/// otherwise ignored.
+	/// `user_id` will be provided back as `user_channel_id` in [`Event::FundingGenerationReady`]
```
Comment (valentinewallace) [inline lightning/src/ln/channelmanager.rs:1275]: Could we rename this parameter to `user_channel_id`?

File: lightning/src/ln/channelmanager.rs [hunk 2]
```diff
@@ -1267,22 +1267,30 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
...
+	/// to allow tracking of which events correspond with which `create_channel` call. Note that
+	/// the `user_channel_id` defaults to 0 for inbound channels, so you may wish to avoid using 0
```
Comment (valentinewallace) [inline lightning/src/ln/channelmanager.rs:1277]: Seems like `user_channel_id` is suited to be an `Option`, which can be `None` for inbound channels

Or an enum
```
UserChannelId {
    Outbound(u64),
    Inbound
}
```

Comment (TheBlueMatt) [reply lightning/src/ln/channelmanager.rs]: Hmm, can we make that a todo in #1125?

Comment (valentinewallace) [reply lightning/src/ln/channelmanager.rs]: Fine by me

File: lightning/src/ln/channelmanager.rs [hunk 3]
```diff
@@ -1292,10 +1293,10 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 	/// one derived from the funding transaction's TXID. If the counterparty rejects the channel
 	/// immediately, this temporary ID will appear in [`Event::ChannelClosed::channel_id`].
 	///
-	/// [`Event::FundingGenerationReady`]: events::Event::FundingGenerationReady
+	/// [`Event::FundingGenerationReady::user_channel_id`]: events::Event::FundingGenerationReady::user_channel_id
 	/// [`Event::FundingGenerationReady::temporary_channel_id`]: events::Event::FundingGenerationReady::temporary_channel_id
 	/// [`Event::ChannelClosed::channel_id`]: events::Event::ChannelClosed::channel_id
-	pub fn create_channel(&self, their_network_key: PublicKey, channel_value_satoshis: u64, push_msat: u64, user_id: u64, override_config: Option<UserConfig>) -> Result<[u8; 32], APIError> {
+	pub fn create_channel(&self, their_network_key: PublicKey, channel_value_satoshis: u64, push_msat: u64, user_channel_id: u64, override_config: Option<UserConfig>) -> Result<[u8; 32], APIError> {
```
Comment (jkczyz) [inline lightning/src/ln/channelmanager.rs:1299]: `git grep user_id` shows some places that need to be updated, especially in docs.

**Review by jkczyz — APPROVED** [review]
Fine with holding off on decision on `channel_user_id` as per #1125.

**Comment by TheBlueMatt** [comment]
Squashed the fixup commit down, diff since Jeff's ACK:

```
$ git diff-tree -U1 347c49e1 bb7ef6c29
diff --git a/fuzz/src/router.rs b/fuzz/src/router.rs
index 6c792916f..b80a8284e 100644
--- a/fuzz/src/router.rs
+++ b/fuzz/src/router.rs
@@ -218,3 +218,3 @@ pub fn do_test<Out: test_logger::Output>(data: &[u8], out: Out) {
 								channel_value_satoshis: slice_to_be64(get_slice!(8)),
-								user_id: 0, inbound_capacity_msat: 0,
+								user_channel_id: 0, inbound_capacity_msat: 0,
 								unspendable_punishment_reserve: None,
$ 
```

**Comment by TheBlueMatt** [comment]
Diff since Val's ACK is also just one trivial variable rename, so gonna take this. Gotta wait on #1127 anyway:

```
$ git diff-tree -U1 539bb6d0aed0c0cc273f78ec3c29b7aeaec64ac0 bb7ef6c2901a9582e19511ab0a974d1de3ef1205
diff --git a/fuzz/src/router.rs b/fuzz/src/router.rs
index 6c792916f..b80a8284e 100644
--- a/fuzz/src/router.rs
+++ b/fuzz/src/router.rs
@@ -218,3 +218,3 @@ pub fn do_test<Out: test_logger::Output>(data: &[u8], out: Out) {
 								channel_value_satoshis: slice_to_be64(get_slice!(8)),
-								user_id: 0, inbound_capacity_msat: 0,
+								user_channel_id: 0, inbound_capacity_msat: 0,
 								unspendable_punishment_reserve: None,
diff --git a/lightning/src/ln/channelmanager.rs b/lightning/src/ln/channelmanager.rs
index 7f4caacfc..0106fe51d 100644
--- a/lightning/src/ln/channelmanager.rs
+++ b/lightning/src/ln/channelmanager.rs
@@ -778,4 +778,4 @@ pub struct ChannelDetails {
 	pub unspendable_punishment_reserve: Option<u64>,
-	/// The user_id passed in to create_channel, or 0 if the channel was inbound.
-	pub user_id: u64,
+	/// The `user_channel_id` passed in to create_channel, or 0 if the channel was inbound.
+	pub user_channel_id: u64,
 	/// The available outbound capacity for sending HTLCs to the remote peer. This does not include
@@ -1363,3 +1363,3 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					outbound_capacity_msat,
-					user_id: channel.get_user_id(),
+					user_channel_id: channel.get_user_id(),
 					confirmations_required: channel.minimum_depth(),
diff --git a/lightning/src/routing/router.rs b/lightning/src/routing/router.rs
index d6adb889d..9e02e10f9 100644
--- a/lightning/src/routing/router.rs
+++ b/lightning/src/routing/router.rs
@@ -1329,3 +1329,3 @@ mod tests {
 			channel_value_satoshis: 0,
-			user_id: 0,
+			user_channel_id: 0,
 			outbound_capacity_msat,
diff --git a/lightning/src/util/events.rs b/lightning/src/util/events.rs
index 8502f9be4..7ea369f55 100644
--- a/lightning/src/util/events.rs
+++ b/lightning/src/util/events.rs
@@ -148,4 +148,4 @@ pub enum Event {
 		output_script: Script,
-		/// The `user_id` value passed in to [`ChannelManager::create_channel`], or 0 for an
-		/// inbound channel.
+		/// The `user_channel_id` value passed in to [`ChannelManager::create_channel`], or 0 for
+		/// an inbound channel.
 		///
@@ -267,4 +267,4 @@ pub enum Event {
 		channel_id: [u8; 32],
-		/// The `user_id` value passed in to [`ChannelManager::create_channel`], or 0 for an
-		/// inbound channel. This will always be zero for objects serialized with LDK versions
+		/// The `user_channel_id` value passed in to [`ChannelManager::create_channel`], or 0 for
+		/// an inbound channel. This will always be zero for objects serialized with LDK versions
 		/// prior to 0.0.102.
$
```

**Pull request merged**